cranelift-wasm/src/func_translator.rs

```diff
@@ -20,15 +20,13 @@ use wasmparser::{self, BinaryReader};
 /// by a `FuncEnvironment` object. A single translator instance can be reused to translate multiple
 /// functions which will reduce heap allocation traffic.
 pub struct FuncTranslator {
-    func_ctx: FunctionBuilderContext,
```

Comment:
This should hold a `FuncBuilder` now, which is cleared between functions, so that it achieves the same effect of holding on to the memory allocations.

Reply:
Hello, sorry I'm only getting back to this now. I was wondering how do I initialise a regular `FunctionBuilder`, as `FuncTranslator` is not given a `Function` until `Function::translate_from_reader` is called?

Reply:
Ah, that's a good question. One option would be to change FunctionBuilder from holding the Function by reference to holding a Function value. Then it could have a method that does `std::replace(&mut self.func, Function::new())` and returns the old Function, so allow consumers to take the Function out when it's done.